base/libgit2/types.jl

```diff
@@ -174,7 +174,50 @@ function RemoteCallbacks(credentials::Ptr{Void}, payload::Ref{Nullable{AbstractC
     RemoteCallbacks(credentials=credentials_cb(), payload=pointer_from_objref(payload))
 end
 
-if LibGit2.version() >= v"0.24.0"
+immutable ProxyOptions
```

Comment:
It's not an API issue. I'm just attempting to roughly match whatever we have on master, not actually redesign that commit.